Libraries/LibWeb/WebIDL/AbstractOperations.cpp

```diff
@@ -275,17 +283,20 @@ JS::Completion invoke_callback(WebIDL::CallbackType& callback, Optional<JS::Valu
         return { JS::js_undefined() };
     }
 
-    // 5. Let relevant realm be F’s associated Realm.
-    auto& relevant_realm = function_object->shape().realm();
+    // 5. Let realm be F’s associated realm.
+    auto& realm = function_object->shape().realm();
 
-    // 6. Let stored realm be value’s callback context.
-    auto& stored_realm = callback.callback_context;
+    // FIXME: 6. Let relevant settings be realm’s settings object.
 
-    // 8. Prepare to run script with relevant realm.
-    HTML::prepare_to_run_script(relevant_realm);
+    // 7. Let stored settings be callable’s callback context.
+    auto& stored_settings = callback.callback_context;
 
-    // 9. Prepare to run a callback with stored realm.
-    HTML::prepare_to_run_callback(stored_realm);
+    // 8. Prepare to run script with relevant settings.
+    // FIXME: This method follows an older version of the spec, which takes a realm, so we use F's associated realm instead.
```

Comment:
I'm easy either way, previously I had tried to update it to the shadow realm merge request but keep the old spec link as well, I think that is what is done in LibJS when we implement proposals

But yeah, this was the main editorial refactor needed for shadow realms since they don't have an ESO. It's basically just a straight replacement of ESO->realm for preparing/cleaning up a script.

Reply:
I've updated the PR to follow https://whatpr.org/webidl/1437.html#invoke-a-callback-function. No functional changes needed.